### ISSUE TITLE
Declaration files should not emit accessor warnings

### ISSUE THREAD

**Issue by weswigham** [microsoft/TypeScript#33641]
In our `docker` suite, some declaration files are issuing errors along the lines of
```
dist-esm/ChangeFeedResponse.d.ts:25:9 - (TS1056) Accessors are only available when targeting ECMAScript 5 and higher.
```
this _might_ be coming from an older version of TS, but if it's not, we need to remove this error, as it doesn't make sense to talk about accessor implementation limitations in declaration files.
